expr_checker.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import ast
+from copy import copy
+
+
+_safe_type = None
+_readonly = None
+
+
+def check_type(obj):
+    if type(obj) in {list, tuple, set}:
+        list(map(check_type, obj))
+        return obj
+    elif type(obj) not in _safe_type:
```

Comment:
can you use `isinstance` instead ?

Reply:
Like this ?

```python
if isinstance(obj, list) or isinstance(obj, tuple) or isinstance(obj, set):
    # blabla

elif any([isinstance(obj, type) for type in _safe_type]):
    #blabla
```

Reply:
isinstance can take a tuple for the second argument
```suggestion
    elif isinstance(obj, _safe_type):
```

(ensure that `_safe_type` is a tuple, it's better than a list / set anyway)

Reply:
I think we should stick with `type` because `isinstance` returns `True` if you inherit of one of the class in the `_safe_type` tuple

For example: 

```python
>>> class Foo:
...     pass
... 
>>> class Bar(Foo):
...     pass
... 
>>> a = Bar()
>>> isinstance(a, Foo)
True
>>> type(a) == Foo
False
>>> 
```

Reply:
I have added a test to prove my point

Reply:
ah ok I see, good point :)